src/pkg/services/m365/api/groups.go

```diff
@@ -111,25 +113,28 @@ func getGroups(
 	return groups, el.Failure()
 }
 
-func FetchOnlyTeams(ctx context.Context, g models.Groupable) bool {
+func FetchOnlyTeams(ctx context.Context, groups []models.Groupable) []models.Groupable {
 	log := logger.Ctx(ctx)
-
-	if g.GetAdditionalData()[ResourceProvisioningOptions] != nil {
-		val, _ := tform.AnyValueToT[[]any](ResourceProvisioningOptions, g.GetAdditionalData())
-		for _, v := range val {
-			s, err := str.AnyToString(v)
-			if err != nil {
-				log.Debug("could not be converted to string value: ", ResourceProvisioningOptions)
-				return false
-			}
-
-			if s == teamsAdditionalDataLabel {
-				return true
+	var teams []models.Groupable
+
+	for _, g := range groups {
+		if g.GetAdditionalData()[ResourceProvisioningOptions] != nil {
+			val, _ := tform.AnyValueToT[[]any](ResourceProvisioningOptions, g.GetAdditionalData())
+			for _, v := range val {
+				s, err := str.AnyToString(v)
+				if err != nil {
+					log.Debug("could not be converted to string value: ", ResourceProvisioningOptions)
+					continue
+				}
+
+				if s == teamsAdditionalDataLabel {
+					teams = append(teams, g)
+				}
```

Comment:
I forget, do we already have this in multiple spots?  If so, can it be moved into a helper func?

Reply:
its used only here and not at multiple places.